sqlglot/dialects/snowflake.py

```diff
@@ -71,6 +71,18 @@ def _unix_to_time(self, expression):
     raise ValueError("Improper scale for timestamp")
 
 
+def _flatten_sql(self, expression):
+    args = [
+        f"{'INPUT' if key == 'this' else key.upper()} => {self.sql(arg)}"
+        for key, arg in expression.args.items()
+    ]
+
+    if self.pretty and self.text_width(args) > self._max_text_width:
+        nl = "\n"
+        return f"FLATTEN({self.indent({nl} + f',{nl}'.join(args) + {nl}, skip_first=True, skip_last=True)})"
+    return f'FLATTEN({", ".join(args)})'
```

Comment:
I was trying to use `format_args` for this, because it's essentially the same logic, but with different sql formatting. Maybe we can dry this up by adding a `sql_lambda` argument to the `format_args` which takes care of the formatting, so we can reuse it for situations like this. For example, perhaps we could do:

```python
    def format_args(self, *args, sql_lambda=None):
        sql_lambda = sql_lambda or self.sql
        args = tuple(sql_lambda(arg) for arg in args if arg is not None)
        if self.pretty and self.text_width(args) > self._max_text_width:
            return self.indent("\n" + f",\n".join(args) + "\n", skip_first=True, skip_last=True)
        return ", ".join(args)
```
So then `_flatten_sql` would become:

```python
def _flatten_sql(self, expression):
    sql_lambda = lambda key_arg: f"{key_arg[0]} => {key_arg[1]}"
    return self.format_args(*[(key, arg) for key, arg in expression.args.items()], sql_lambda=sql_lambda)
```